client.go

```diff
@@ -74,15 +85,26 @@ func New(c Config) Client {
 		c.RetryConfig,
 	)
 
+	if c.APIBaseURL != "" {
+		return NewClientWithBaseUrl(httpClient, c.APIBaseURL)
+	}
+
 	return NewClient(httpClient)
 }
 
-// NewClient creates a new Site24x7 API Client from httpClient. This can be
-// used to provide a custom http client for use with the API. The custom http
+// NewClient creates a new Site24x7 API Client from httpClient with default API base URL.
+// This can be used to provide a custom http client for use with the API. The custom http
 // client has to transparently handle the Site24x7 OAuth flow.
 func NewClient(httpClient HTTPClient) Client {
+	return NewClientWithBaseUrl(httpClient, APIBaseURL)
+}
+
+// NewClientWithBaseUrl creates a new Site24x7 API Client from httpClient and given API base URL.
+// This can be used to provide a custom http client for use with the API. The custom http
+// client has to transparently handle the Site24x7 OAuth flow.
+func NewClientWithBaseUrl(httpClient HTTPClient, baseUrl string) Client {
 	return &client{
-		restClient: rest.NewClient(httpClient, APIBaseURL),
+		restClient: rest.NewClient(httpClient, baseUrl),
```

Comment:
Please change `NewClientWithBaseUrl` and `baseUrl` to `NewClientWithBaseURL` and `baseURL` respectively to make the usage of `URL` consistent. Acronyms should be all-caps or all-lowercase: https://github.com/golang/go/wiki/CodeReviewComments#initialisms

Reply:
Acronyms all-capsed. Total Go newbie, so thanks for pointing out the conventions.